server/artifacts/artifact_server.go

```diff
@@ -128,6 +128,9 @@ func (a *ArtifactServer) returnArtifact(ctx context.Context, w http.ResponseWrit
 	kubeClient := auth.GetKubeClient(ctx)
 
 	art := wf.Status.Nodes[nodeId].Outputs.GetArtifactByName(artifactName)
+	if art == nil {
+		art = wf.Status.Nodes[nodeId].Inputs.GetArtifactByName(artifactName)
+	}
```

Comment:
I think a better approach might be adding `inputs/outputs` as part of the URL parameters.

For example, Instead of:

```
/artifacts/argo/artifact-passing-fxng8/artifact-passing-fxng8-3010801439/message
```

We can add `input/` before the artifact name.

```
/artifacts/argo/artifact-passing-fxng8/artifact-passing-fxng8-3010801439/input/message
```

Happy to implement this too. But want to hear some thoughts first.

Reply:
I think the correct solution would a new end point for input artifacts.  I’m interested to understand when users would want to see inputs mind you.

Reply:
Thanks for the feedback @alexec!

> I’m interested to understand when users would want to see inputs mind you.

I think people use input artifact links mostly for convenience.

For example, an organization may have a big workflow with multiple different steps gathering data from the different data source. Each step is maintained by a different team.  There is a final step to aggregate all the data. It's easier for developers to click that final step and view all the input artifacts instead of finding them in the previous steps' outputs.

> I think the correct solution would a new endpoint for input artifacts.

I wonder if we are talking about something like this:

```diff
        mux.HandleFunc("/api/", func(w http.ResponseWriter, r *http.Request) { webhookInterceptor(w, r, gwmux) })
-       mux.HandleFunc("/artifacts/", artifactServer.GetArtifact)
-       mux.HandleFunc("/artifacts-by-uid/", artifactServer.GetArtifactByUID)
+       mux.HandleFunc("/artifacts/input/", artifactServer.GetInputArtifact)
+       mux.HandleFunc("/artifacts/output/", artifactServer.GetOutputArtifact)
+       mux.HandleFunc("/artifacts-by-uid/input/", artifactServer.GetInputArtifactByUID)
+       mux.HandleFunc("/artifacts-by-uid/output/", artifactServer.GetOutputArtifactByUID)
```

The current URL pattern is `/artificats/{namespace}/{workflowName}/{nodeID}/{artifactName}`, another option is to put `input`/`output` between `nodeID` and `artifactName`: `/artificats/{namespace}/{workflowName}/{nodeID}/input/{artifactName}`

Let me know what you think. Thanks!

Reply:
We should keep the old URL otherwise it might break users.

```
/output-artifacts/{namespace}/{workflowName}/{nodeID}/{artifactName}
/output-artifacts-by-uid/{namespace}/{uuid}/{nodeID}/{artifactName}
```